### DISCUSSION TITLE
I'm using live mode and I'm passing PCM data in the stream function and it's reading fail: EOF, what should I do?

### DISCUSSION THREAD

**Discussion by gouyuwang** [deepgram/community#490] [General help] [answered]
`
 dclient := client.New(ctx, d.token, &client.ClientOptions{}, interfaces.LiveTranscriptionOptions{
 ***
dclient.Stream(bytes.NewReader(data)) // data is PCM
`

response:

I1212 09:49:01.305190   90294 client.go:269] r.Read failed. Err: EOF

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment, marked as answer]
Hi @gouyuwang, when sending PCM data, make sure to include the `encoding` and `sample_rate` parameters in the LiveTranscriptionOptions. See [our docs](https://developers.deepgram.com/reference/streaming) for how to set those parameters.

**Reply by gouyuwang** [reply]
thanks！Solved.
